File: src/lib/library/v1/color.ts
import { GpuDesignerNode } from "../../designer/gpudesignernode";
import { Color } from "@/lib/designer/color";

export class ColorizeNode extends GpuDesignerNode {
	public init() {
		this.title = "Colorize";

		this.addInput("image");

		this.addColorProperty("color", "Color", new Color());

		const source = `
        vec4 process(vec2 uv)
        {
            return texture(image,uv) * prop_color;
        }
        `;

		this.buildShader(source);
	}
}

export class ColorNode extends GpuDesignerNode {
	public init() {
		this.title = "Color";

		this.addColorProperty("color", "Color", new Color());

		const source = `
        vec4 process(vec2 uv)
        {
            return prop_color;
        }
        `;

		this.buildShader(source);
	}
}

File: src/texturelab/libraries/v2/color.cpp
#include "../../models.h"
#include "../../props.h"
#include "../libv2.h"

void ColorNode::init()
{
    this->title = "Color";

    this->addInput("image");

    // todo: add props
    this->setShaderSource(R""""(
        vec4 process(vec2 uv)
        {
            vec4 prop_color = vec4(1,1,1,1);
            return texture(image,uv) * prop_color;
            //return vec4(uv, vec2(0.0,1.0));// * prop_color;
        }
        )"""");
}